webapp/src/components/MessageList.tsx

```diff
@@ -0,0 +1,62 @@
+'use client';
+
+import ListItem from '@mui/material/ListItem';
+import { FC, useCallback } from 'react';
+import { FixedSizeList, ListChildComponentProps } from 'react-window';
```

Comment:
Good comment! The designs include a custom search feature so that shouldn't be a problem long term. I also think we should do what some other apps (eg Google Docs) do which is to override the browser search keyboard shortcut to focus the custom search feature.

Reply:
Yeah the custom search and the shortcut override was exactly what I was thinking. In order to keep the built-in search while also preserving performance, the only option I can see would be to ditch React, render everything as plain HTML, and then add the interactivity on top using small amounts of JS. It's actually a really nice way to build stuff too, but a massive departure from industry trends and Zetkin's usual React setup.

Reply:
Yeah lets not get carried away. 😊